projects/dsp-ui/src/lib/viewer/values/text-value/text-value-as-xml/text-value-as-xml.component.ts

```diff
@@ -0,0 +1,209 @@
+import { Component, Inject, Input, OnChanges, OnDestroy, OnInit, SimpleChanges } from '@angular/core';
+import { BaseValueComponent } from '../../base-value.component';
+import { Constants, CreateTextValueAsXml, ReadTextValueAsXml, UpdateTextValueAsXml } from '@dasch-swiss/dsp-js';
+import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { ValueErrorStateMatcher } from '../../value-error-state-matcher';
+import * as Editor from 'ckeditor5-custom-build';
+import { AppInitService } from '../../../../core';
+
+@Component({
+    selector: 'dsp-text-value-as-xml',
+    templateUrl: './text-value-as-xml.component.html',
+    styleUrls: ['./text-value-as-xml.component.scss']
+})
+export class TextValueAsXMLComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+
+    @Input() displayValue?: ReadTextValueAsXml;
+    @Input() mapping = 'http://rdfh.ch/standoff/mappings/StandardMapping'; // TODO: define this somewhere else
+
+    valueFormControl: FormControl;
+    commentFormControl: FormControl;
+
+    form: FormGroup;
+
+    valueChangesSubscription: Subscription;
+    matcher = new ValueErrorStateMatcher();
+    customValidators = [];
+
+    // https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/frameworks/angular.html
+    editor: Editor;
+    editorConfig;
+
+    // TODO: get this from config via AppInitService
+    readonly resourceBasePath = 'http://rdfh.ch/';
+
+    constructor(private _appInitService: AppInitService,
+                @Inject(FormBuilder) private fb: FormBuilder) {
+        super();
+    }
+
+    getInitValue(): string | null {
+
+        // check for standard mapping
+        if (this.displayValue !== undefined) {
+
+            // strip the doctype and text tag
+            return this._handleXML(this.displayValue.xml, true);
+        } else {
+            return null;
+        }
+    }
+
+    ngOnInit() {
+
+        if (this.mapping !== undefined
+            && this._appInitService.config['xmlTransform'] !== undefined
```

Comment:
The linter gives me these strange warnings but I think it's okay.
<img width="1044" alt="Screen Shot 2020-09-04 at 09 45 39" src="https://user-images.githubusercontent.com/60604010/92213192-6a585b00-ee93-11ea-983c-7b8b2cfbd3ee.png">

Reply:
yes, I thought this could be improved by passing in a type for the config object when accessing the app init service, see https://dasch.myjetbrains.com/youtrack/issue/DSP-601

so then we could get access the members of an object